tests/integration_rust/install_tests.rs

```diff
@@ -794,3 +794,60 @@ async fn pypi_prefix_is_not_created_when_whl() {
     // Check that the prefix is not created
     assert!(!default_env_prefix.exists());
 }
+
+/// This test checks that the override of a conda package is correctly done per platform.
+/// There have been issues in the past that the wrong repodata was used for the override.
+/// What this test does is recreate this situation by adding a conda package that is only
+/// available on linux and then adding a PyPI dependency on the same package for both linux
+/// and osxarm64.
+/// This should result in the PyPI package being overridden on linux and not on osxarm64.
+#[tokio::test]
+async fn conda_pypi_override_correct_per_platform() {
+    let pixi = PixiControl::new().unwrap();
+    pixi.init_with_platforms(vec![
+        Platform::OsxArm64.to_string(),
+        Platform::Linux64.to_string(),
```

Comment:
```suggestion
        Platform::Linux64.to_string(),
        Platform::Win64.to_string(),
        Platform::Osx64.to_string(),
```

I guess this should do it.